packages/webhooks/model-validator/validator.js

```diff
@@ -0,0 +1,208 @@
+        this.models = []
+        this.filterValidators = {}
+        this.fieldsValidators = {}
+    }
+
+    /**
+     * Registers GraphQL model from schema in internal base
+     * Also generates JSON-schema validators for validating its fields and filters
+     * @param {string} modelName
+     */
+    registerModel (modelName) {
+        if (this.models.includes(modelName)) return
+
+        this.models.push(modelName)
+
+        const modelFilterSchema = this.generator.getSchemaForSymbol(`${modelName}WhereInput`)
+        const ajv = new Ajv({ strict: true, allErrors: true })
+        ajv.validateSchema(modelFilterSchema)
+        if (ajv.errors) {
+            throw new Error(`Schema for class ${modelName}WhereInput was not valid!`)
+        }
+        this.filterValidators[modelName] = ajv.compile(modelFilterSchema)
+
+        const modelSchema = this.generator.getSchemaForSymbol(modelName)
+        ajv.validateSchema(modelSchema)
+        if (ajv.errors) {
+            throw new Error(`Schema for class ${modelName} was not valid!`)
+        }
+        const maskedSchema = WebHookModelValidator._maskSchemaTypes(modelSchema)
+        ajv.validateSchema(maskedSchema)
+        if (ajv.errors) {
+            throw new Error(`Masked schema for class ${modelName} was not valid!`)
+        }
+        this.fieldsValidators[modelName] = ajv.compile(maskedSchema)
+    }
+
+    /**
+     * Validates specified model filters
+     * @param {string} modelName
+     * @param {JSON} filters
+     * @returns {{isValid: boolean, errors: *[]}}
+     */
+    validateFilters (modelName, filters) {
+        if (!this.models.includes(modelName)) {
+            throw new Error(`Unregistered model name: ${modelName}`)
+        }
+        const validator = this.filterValidators[modelName]
+        const isValid = validator(filters)
+        if (isValid) {
+            return { isValid, errors: [] }
+        } else {
+            return { isValid, errors: validator.errors }
+        }
+    }
+
+    /**
+     * Validates normalized field-string (Subset of model fields in GQL notation)
+     * @param modelName
+     * @param {string} modelName
+     * @param {string} fieldString
+     * @returns {{isValid: boolean, errors: string[]}|{isValid, errors}|{isValid, errors: *[]}}
+     */
+    validateFields (modelName, fieldString) {
+        if (!this.models.includes(modelName)) {
+            throw new Error(`Unregistered model name: ${modelName}`)
+        }
+        const validator = this.fieldsValidators[modelName]
+        const { data, error } = WebHookModelValidator._buildSampleFromFieldsString(fieldString)
+        if (error) {
+            return { isValid: false, errors: [error] }
+        } else {
+            const isValid = validator(data)
+            if (isValid) {
+                return { isValid, errors: [] }
+            } else {
+                return { isValid, errors: validator.errors }
+            }
+        }
```

Comment:
```js
const isValid = !error && validator(data)

if (isValid) {
    return { isValid, errors: [] }
} else {
    ...
}

```

Reply:
Throwing error explicitly is more preferable then explicitly feeding `null` value to validator